src/which.js

```diff
@@ -45,7 +40,17 @@ function _which(options, cmd) {
   var where = null;
 
   // No relative/absolute paths provided?
-  if (cmd.search(/\//) === -1) {
+  if (cmd.indexOf('/') === -1) {
+    // Assume that there are no extensions to append to queries (this is the
+    // case for unix)
+    var pathExtArray = [''];
+    if (common.platform === 'win') {
+      // In case the PATHEXT variable is somehow not set (e.g.
+      // child_process.spawn with an empty environment), use the XP default.
+      var pathExtEnv = process.env.PATHEXT || XP_DEFAULT_PATHEXT;
+      pathExtArray = splitPath(pathExtEnv.toUpperCase());
+    }
+
     // Search for command in PATH
     pathArray.forEach(function (dir) {
```

Comment:
Here, we're looping through the path array, and setting `where` under certain conditions, then returning from the `forEach` callback. On the next call, we check if `which` was set, and if it was, we don't continue. FInally, at the end of `checkPath`, we return `where` if it was set. To me, it seems like a regular `for` loop would be a better choice here, because we could return `where` from the function immediately.